src/App.tsx

```diff
@@ -1,41 +1,62 @@
-import React from 'react';
+/* eslint-disable no-console */
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { getUserPosts } from './api/posts';
+import { UserSelect } from './components/UserSelect/UserSelect';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
-
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+const App: React.FC = () => {
+  const [user, setUser] = useState<number | string>('all');
+  const [posts, setPosts] = useState([]);
+  const [selectedPostId, setSelectedPostId] = useState<number | null>(null);
+
+  useEffect(() => {
+    getUserPosts(user)
+      .then(data => {
+        setPosts(data);
+      });
```

Comment:
```suggestion
      .then(setPosts);
```
Можна використати коротший синтаксис

Reply:
о, це круто. Супер-коротко, хоча не дуже explanatory. 
Використовуватиму.